examples/run_quickstart_new_config.py

```diff
@@ -0,0 +1,156 @@
+
+import os
+from typing import Any
+
+import pandas as pd
+
+from tabrepo import EvaluationRepository, EvaluationRepositoryCollection, Evaluator
+from tabrepo.benchmark.experiment import AGModelBagExperiment, Experiment, ExperimentBatchRunner
+
+
+if __name__ == '__main__':
+    # Load Context
+    context_name = "D244_F3_C1530_10"  # 10 smallest datasets. To run larger, set to "D244_F3_C1530_200"
+    expname = "./experiments/quickstart_new_config"  # folder location to save all experiment artifacts
+    ignore_cache = True  # set to True to overwrite existing caches and re-run experiments from scratch
+
+    # The original TabRepo artifacts for the 1530 configs
+    repo_og: EvaluationRepository = EvaluationRepository.from_context(context_name, cache=True)
+    datasets = repo_og.datasets()
+
+    # Sample for a quick demo
+    datasets = datasets[:3]
+    folds = [0]
+
+    # To run everything:
+    # datasets = repo_og.datasets()
+    # folds = repo_og.folds
+
+    # import your model classes (can be custom, must inherit from AbstractModel)
+    from autogluon.tabular.models import LGBModel, XGBoostModel
+
+    # import your non-autogluon model classes (must inherit from AbstractExecModel)
+    from tabrepo.benchmark.models.simple import SimpleLightGBM
+
+    # This list of methods will be fit sequentially on each task (dataset x fold)
+    methods = [
+        # This will be a `config` in EvaluationRepository, because it computes out-of-fold predictions and thus can be used for post-hoc ensemble.
+        AGModelBagExperiment(  # Wrapper for fitting a single bagged model via AutoGluon
+            # The name you want the config to have
+            name="LightGBM_c1_BAG_L1_Reproduced",  # Reproduces the `LightGBM_c1_BAG_L1` model from the TabRepo 2024 paper
+
+            # The class of the model. Can also be a string if AutoGluon recognizes it, such as `"GBM"`
+            # Supports any model that inherits from `autogluon.core.models.AbstractModel`
+            model_cls=LGBModel,  # model_cls="GBM",  <- identical
+            model_hyperparameters={},  # The non-default model hyperparameters.
+            num_bag_folds=8,  # num_bag_folds=8 was used in the TabRepo 2024 paper
+            time_limit=3600,  # time_limit=3600 was used in the TabRepo 2024 paper
+        ),
+        AGModelBagExperiment(
+            # Reproduces the `XGBoost_c1_BAG_L1` model from the TabRepo 2024 paper
+            # The results may slightly differ since we are using a different XGBoost version.
+            name="XGBoost_c1_BAG_L1_Reproduced",
+            model_cls=XGBoostModel,
+            model_hyperparameters={},
+            num_bag_folds=8,
+            time_limit=3600,
+        ),
+
+        # This will be a `baseline` in EvaluationRepository, because it doesn't compute out-of-fold predictions and thus can't be used for post-hoc ensemble.
+        Experiment(  # Generic wrapper for any model, including non-AutoGluon models
```

Comment:
Could we make the naming more precise here and similar to the one above? Maybe `AGModelSingleExperiment`?

Reply:
Experiment is fully generic, and is not tied to AG in any way, so I prefer to keep the name separated from AG.

I already have `AGModelExperiment` for generic AG runs. `Experiment` is the most generic as it could be anything (for example, H2OAutoML or native TabPFNv2)